openlibrary/components/AuthorIdentifiers.vue

```diff
@@ -29,12 +29,24 @@
 </template>
 
 <script>
+const uuidV4Validator = /^[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}$/
+const genericNumericValidator = /^[0-9]+$/
 const identifierPatterns  = {
     wikidata: /^Q[1-9]\d*$/i,
     isni: /^[0]{4} ?[0-9]{4} ?[0-9]{4} ?[0-9]{3}[0-9X]$/i,
     lc_naf: /^n[bors]?[0-9]+$/,
     amazon: /^B[0-9A-Za-z]{9}$/,
     youtube: /^@[A-Za-z0-9_\-.]{3,30}/,
+    viaf: genericNumericValidator,
+    bookbrainz: uuidV4Validator,
+    musicbrainz: uuidV4Validator,
+    imdb: /^\w{2}\d+$/,
+    goodreads: genericNumericValidator,
+    librarything: /^\S+$/,
+    librivox: genericNumericValidator,
+    project_gutenberg: /^[1-9]\d{0,4}$/,
```

Comment:
Pretty sure Gutenberg IDs can be more than 5 digits long (there just isn’t more than 100k authors currently; the most recently added ones seem to be in the late 50k’s?). Not sure if it makes sense to go `/^[1-9]\d{0,5}$/` for future proofing for the foreseeable future (up to 1 million Gutenberg authors!), while still not allowing almost certainly false ids (like 2931870911 or something), or simply to do `\d*` which would never needs updating.
```suggestion
    project_gutenberg: /^[1-9]\d{0,5}$/,
```
of course, since this seems similar, if not identical, to the `genericNumericValidator`, it might also make sense to just reuse that… :)
```suggestion
    project_gutenberg: genericNumericValidator,
```

Reply:
I'll take your word for these - I just pulled them directly from how Wikidata validates these particular identifiers!